### ISSUE TITLE
[binding-coap] Name conflict "interface" in coap@0.21.0 causes problem

### ISSUE THREAD

**Issue by adriankast** [eclipse/thingweb.node-wot#137]
# Setup
* node 10.16.3
* npm 6.12.1
* Windows 10 (but also found on macOS and Linux)
* Vue project with electron-builder addon
* @node-wot/binding-coap 0.6.2 (also found with 0.6.3-SNAPSHOT.2)

# Doing
* npm install
* npm run electron:build (calls vue-dev tool script that launches the electron-builder in production mode)

# Problem
The build fails with the error message:
> ERROR  Failed to compile with 1 errors 
>                                                                                                                                               
> js/chunk-vendors.a6cfe1de.js from Terser
> Unexpected token: name (interface) [./node_modules/coap/lib/server.js:230,0][js/chunk-vendors.a6cfe1de.js:18868,58]

# Notes
The Problem appears in Version 0.21.0 of the coap package and is already fixed in 0.22.0, so updating the package to 0.22.0 should fix the problem. It is fixed in the coap package in this [commit](https://github.com/mcollina/node-coap/commit/0c5d52b787826e6cfeff7ed140dfbe442f6c6f7c#diff-c945a46d13b34fcaff544d966cffcaba).

# Workaround
Correct the name conflict with "interfaces" in the coap module ./node_modules/coap/lib/server.js line 230 and line 231(two words) by renaming:

> interface

to 

> _interface

or any other valid expression that is not reserved/shadowed.  
If you need to automate it you can use:
* Powershell
```
    - (Get-Content -path .\node_modules\coap\lib\server.js) -creplace '\(interface\)','(_interface)' | Set-Content -Path .\node_modules\coap\lib\server.js
    - (Get-Content -path .\node_modules\coap\lib\server.js) -creplace ', interface\)',', _interface)' | Set-Content -Path .\node_modules\coap\lib\server.js
```
* Shell
```
sed -i 's/(interface)/(_interface)/g' ./node_modules/coap/lib/server.js
sed -i 's/, interface)/, _interface)/g' ./node_modules/coap/lib/server.js
```

**Comment by danielpeintner** [comment]
I updated the coap dependency to the latest version `0.23.1` (see https://github.com/eclipse/thingweb.node-wot/commit/ca50a2df47c5032467ceefe426e05f0185440f2b).

Please reopen it it still not resolves the issue.

Note: If you depend on the API version 0.6.x we need to apply this fix also to the [v0.6.x](https://github.com/eclipse/thingweb.node-wot/tree/v0.6.x) branch.

**Comment by adriankast** [comment]
Just checked with `binding-coap` Version `0.7.0-SNAPSHOT.2` at least on my setup the Issue is solved ✔️
